### PR TITLE
Support account link transaction

### PR DESCRIPTION
**Pull request by NemTechCN**
issue #28

### CODE REVIEW

File: src/commands/transaction/accountlink.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+export class CommandOptions extends ProfileOptions {
+    @option({
+        flag: 'p',
+        description: '<remoteaccountkey>',
+    })
+    publickey: string;
+    @option({
+        flag: 'a',
+        description: '< 0 (add)| 1 (delete)>',
```
Comment (rg911) [inline src/commands/transaction/accountlink.ts]: Should be `Link` / `Unlink`

Comment (NemTechCN) [reply src/commands/transaction/accountlink.ts]: ok

File: src/commands/transaction/accountlink.ts [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        description: '<remoteaccountkey>',
+    })
+    publickey: string;
+    @option({
+        flag: 'a',
+        description: '< 0 (Link)| 1 (Unlink)>',
+    })
+    action: number;
+
+    validateAction(value: number): 0|1 {
```
Comment (dgarcia360) [inline src/commands/transaction/accountlink.ts]: As we are returning the same value we are passing, we could write something like:

```
if([0,1].some(x => (x === value))) return value;
throw new ExpectedError('Introduce a valid action value');

```

Comment (NemTechCN) [reply src/commands/transaction/accountlink.ts]: This looks good

File: src/commands/transaction/accountlink.ts (hunk 1 above)
Comment (dgarcia360) [reply src/commands/transaction/accountlink.ts]: @rg911 I don't think we need to check in the cli if the account virginity is considered. The WebSocket will return an error anyway.

File: src/commands/transaction/accountlink.ts [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+})
+export default class extends ProfileCommand {
+
+    constructor() {
+        super();
+    }
+
+    @metadata
+    execute(options: CommandOptions) {
+        const profile = this.getProfile(options);
```
Comment (dgarcia360) [inline src/commands/transaction/accountlink.ts]: If the user does not set the options at the beginning, the program should ask to add them.

Example: Introduce the public key of the remote account

File: src/commands/transaction/accountlink.ts [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        description: 'The public key of the remote account',
+    })
+    publickey: string;
+    @option({
+        flag: 'a',
+        description: 'Alias action (0: Add, 1: Remove)',
+    })
+    action: number;
+
+    validateAction(value: number) {
```
Comment (dgarcia360) [inline src/commands/transaction/accountlink.ts]: Think about using a validator. Example: https://github.com/nemtech/nem2-cli/blob/master/src/commands/profile/create.ts#L47

The validator should be tested.

**Comment by dgarcia360** [comment]
Hi @NemTechCN, do you plan to continue with this PR?

**Comment by NemTechCN** [comment]
yes @dgarcia360

**Pull request merged**